### PR TITLE
Upgrade github.com/google/go-containerregistry to latest to fix 404s

### PR DESCRIPTION
**Pull request by hbd**
This fixes issue #56.

### CODE REVIEW

**Comment by hbd** [comment]
I'm having a hard time understanding the test failure output. Could someone more familiar with the tests please advise? cc @vito @osis

**Comment by hbd** [comment]
This looks similar to #52, but with an up to date `go.mod` (#52 has conflicts).

**Comment by vito** [comment]
> I'm having a hard time understanding the test failure output. Could someone more familiar with the tests please advise? cc @vito @osis

Something about the new code is probably raising errors differently than how we used to check for them:

https://github.com/concourse/registry-image-resource/blob/28e8085a68dc558af294006441f766b6eb731de5/cmd/check/main.go#L122-L133

**Comment by vito** [comment]
Actually, maybe not, it's actually failing before it even gets there:

https://github.com/concourse/registry-image-resource/blob/28e8085a68dc558af294006441f766b6eb731de5/cmd/check/main.go#L60

**Comment by vito** [comment]
I think we just get that error earlier than we used to, actually. It looks like `remote.Image` actually tries to fetch the manifest now, whereas before that was deferred to when we call `.Digest` on the image.

**Comment by hbd** [comment]
I have moved the manifest check to happen after retrieving the image instead of after the Digest (as the image will not exist if the manifest does not exist). This should fix any outstanding problems.

File: cmd/in/main.go [hunk 1]
```diff
@@ -170,14 +170,8 @@ func rootfsFormat(dest string, req InRequest, image v1.Image) {
 	}
 
 	env := cfg.Config.Env
-	if len(env) == 0 {
```
Comment (vito) [inline cmd/in/main.go]: What was the cue for removing these? :thinking: 

It looks like the tests pass either way, but they also passed before, so I wonder whether this was important and if we need test coverage for it or if I was just kind of guessing that it'd be useful.

Comment (hbd) [reply cmd/in/main.go]: `config.ContainerConfig` was removed in [this commit](https://github.com/google/go-containerregistry/commit/2be26e624e32c4b3de62287e153e5e382a79b7c5), and moved to `/pkg/legacy/config. LayerConfigFile`. I didn't find a reference to `LayerConfigFile` here, so I removed the reference to `ContainerConfig` altogether.

Comment (vito) [reply cmd/in/main.go]: Good enough for me. If someone complains we can just introduce legacy compatibility and add a test. :+1:

**Pull request merged**